src/Coupon/SyncerHooks.php

```diff
@@ -277,7 +279,7 @@ protected function get_coupon_to_delete( WC_Coupon $coupon ): WCCouponAdapter {
 	 * @param int $coupon_id
 	 */
 	protected function handle_delete_coupon( int $coupon_id ) {
-		if ( $this->notifications_service->is_ready() ) {
+		if ( $this->notifications_service->is_ready( self::get_coupons_datatype() ) ) {
```

Comment:
Same here we should use `$this->` instead of `self::`

Reply:
Thanks. Fixed here https://github.com/woocommerce/google-listings-and-ads/pull/2804/commits/e595f4c28c4832b84f9f0079e3c9554aeac82124